### PR TITLE
Ps 24 fix permissions

### PR DESCRIPTION
**Pull request by pivnicek**
running this as root again for apache.

### CODE REVIEW

**Comment by odinuv** [comment]
how can the sandbox user then read the files when they're created by root?

**Comment by pivnicek** [comment]
@odinuv the gid for everything created in the datadir is set by the sandbox and the datadir is 7s regardless.

**Comment by odinuv** [comment]
yeah, so to answer my question it's the magic `+s` in chmod, i'm still slightly worried about the owner, but if it works, it works

**Review by odinuv — APPROVED** [review]

**Comment by pivnicek** [comment]
Actually, I shouldn't have reverted that last commit, that's the one that works best, run the loading as user and apache as root.

**Comment by pivnicek** [comment]
Sorry, I changed it again.  It's all finally working

File: src/run.php [hunk 1]
```diff
@@ -62,6 +62,7 @@
     } else {
         $log->info('There are no input tables.', ['runId' => $runId]);
     }
+    $fs->chmod($validator->getDataDir(), 0777, 0000, true);
```
Comment (odinuv) [inline src/run.php:66]: for readability, can you put this right after the mkdirs ?

File: Dockerfile [hunk 2]
```diff
@@ -8,6 +8,7 @@ RUN apt-get update && apt-get install -y \
 	    git \
         libzip-dev \
         unzip \
+        sudo \
```
Comment (odinuv) [inline Dockerfile]: is needed?

**Review by odinuv — APPROVED** [review]

File: src/run.php (hunk 1 above)
Comment (pivnicek) [reply src/run.php:66]: Not really, because it proceeds to make files and scripts and I need those to all have permissions

Comment (pivnicek) [reply src/run.php:66]: @odinuv I added a comment there that maybe helps.

Comment (odinuv) [reply src/run.php:66]: hmm, so if the rstudio/jupyter creates a file, will the data-loader be able to read it ?

Comment (pivnicek) [reply src/run.php:66]: no, but it doesn't need to, it only runs on start up.  The API can read the files though

File: Dockerfile (hunk 2 above)
Comment (pivnicek) [reply Dockerfile]: no, sorry, I was using it but not anymore

File: src/run.php (hunk 1 above)
Comment (odinuv) [reply src/run.php:66]: right! data-loader doesn't have to, but data-loader-api has to

**Pull request merged**